newsfragments/2675.fixed.md

```diff
@@ -0,0 +1 @@
+`impl FromPyObject for [T; N]` will accept anything passing `PySequence_Check`, e.g. NumPy arrays, in the same way that `impl FromPyObject for Vec<T>` was modified after [#2477](https://github.com/PyO3/pyo3/pull/2477).
```

Comment:
Given this is a revert, I think it'd be quite handy to state what version the behaviour is reverting to rather than the offending PR. Users who want to see the PR can start with the link from the CHANGELOG to this PR and follow the discussion thread.

E.g. "Fix regression of `impl FromPyObject for [T; N]` no longer accepting types passing `PySequence_Check`, e.g. NumPy arrays, since 0.17.0."

Reply:
Adjusted the wording, please re-review.